src/modules/languages/c.nix

```diff
@@ -6,16 +6,29 @@ in
 {
   options.languages.c = {
     enable = lib.mkEnableOption "tools for C development";
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.stdenv.cc;
+      example = "pkgs.clangStdenv.cc";
+      defaultText = "pkgs.stdenv.cc";
+      description = "C/C++ toolchain to use. Defaults to nixpkgs default C compiler (i.e. gcc on Linux and clang on macOS)";
+    };
+    languageServer = lib.mkOption {
+      type = lib.types.nullOr lib.types.package;
+      default = pkgs.clang-tools;
+      example = pkgs.ccls;
+      defaultText = "pkgs.clang-tools";
+      description = "Language server to use. Defaults to clangd from the clang-tools package.";
```

Comment:
Clangd is better maintained and also maintained by the original author of ccls. It scales very well also to large projects like the linux kernel.